From 0b0e5fa9a00ad06e596eb85da04f9f0da561ecae Mon Sep 17 00:00:00 2001
From: FlameFlag <git@ps1.sh>
Date: Wed, 22 Apr 2026 18:27:46 +0300
Subject: [PATCH] BitmapComboBox: use m_bitmapbundles (wxWidgets 3.1.7+) in
 OnDrawItem

wxWidgets 3.1.7 replaced wxBitmapComboBoxBase::m_bitmaps with
m_bitmapbundles (wxVector<wxBitmapBundle>) to support DPI-aware
bitmap bundles. Use GetBitmapFor(this) to extract a wxBitmap sized
for the target window.
---
 src/slic3r/GUI/BitmapComboBox.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/slic3r/GUI/BitmapComboBox.cpp b/src/slic3r/GUI/BitmapComboBox.cpp
index 36c1534..4722a8a 100644
--- a/src/slic3r/GUI/BitmapComboBox.cpp
+++ b/src/slic3r/GUI/BitmapComboBox.cpp
@@ -131,7 +131,7 @@ void BitmapComboBox::OnDrawItem(wxDC& dc,
     int item,
     int flags) const
 {
-    const wxBitmap& bmp = *(static_cast<wxBitmap*>(m_bitmaps[item]));
+    wxBitmap bmp = m_bitmapbundles[item].GetBitmapFor(this);
     if (bmp.IsOk())
     {
         // we should use scaled! size values of bitmap
-- 
2.53.0

